pkg/validation/options.go

```diff
@@ -281,6 +281,11 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 	case *providers.GitLabProvider:
 		p.AllowUnverifiedEmail = o.InsecureOIDCAllowUnverifiedEmail
 		p.Groups = o.GitLabGroup
+		p.Projects = o.GitlabProjects
+		// TODO (@NickMeves): merge groups and projects when calling allowed groups - Refactor to Authorize
+		p.SetAllowedGroups(o.GitlabProjects)
```

Comment:
Like I mentioned in the other comment, we should prefix this.

We'll also need to handle multiple things being added via `SetAllowedGroups`.

Maybe a helper in the provider to take the Groups & Projects and spit out a list of both together with the `group:<GROUP>` and `project:<PROJECT>` prefixes.

Thinking out loud, do Groups need a `group:` prefix? Or do we only prefix things that are group-like but not explicitly named group in cases where we have multiple potential group like entries?

Reply:
I think explicit is better than implicit so I'm more in favor of an harmonization of `kind:value`

Reply:
https://github.com/oauth2-proxy/oauth2-proxy/pull/630/commits/f8186bf2fb16c6cc3ab80936af9cea7edb82a2a1